cclib/method/ddec.py

```diff
@@ -88,17 +96,34 @@ def _read_proatom(self, directory, atom_num, charge):
         #       atoms.h5
         # File format:
         #   Starting from line 13, each line contains the charge densities for each shell
-        chargemol_path = os.path.join(
+        charge_floor = int(math.floor(charge))
+        charge_ceil = int(math.ceil(charge))
+        
+        chargemol_path_floor = os.path.join(
+            directory,
+            "c2_{:03d}_{:03d}_{:03d}_500_100.txt".format(atom_num, atom_num, atom_num - charge_floor),
+        )
+        chargemol_path_ceil = os.path.join(
             directory,
-            "c2_{:03d}_{:03d}_{:03d}_500_100.txt".format(atom_num, atom_num, atom_num - charge),
+            "c2_{:03d}_{:03d}_{:03d}_500_100.txt".format(atom_num, atom_num, atom_num - charge_ceil),
         )
         horton_path = os.path.join(directory, "atoms.h5")
 
-        if os.path.isfile(chargemol_path):
+
+        if os.path.isfile(chargemol_path_floor) or os.path.isfile(chargemol_path_ceil):
             # Use chargemol proatom densities
             # Each shell is .05 angstroms apart (uniform).
             # *scalefactor* = 10.58354497764173 bohrs in module_global_parameter.f08
-            density = numpy.loadtxt(chargemol_path, skiprows=12, dtype=float)
+            if atom_num <= charge_floor:
+                density_floor = numpy.array([0])
+            else:
+                density_floor = numpy.loadtxt(chargemol_path_floor, skiprows=12, dtype=float)
+            if atom_num <= charge_ceil:
```

Comment:
Without my understanding of where this comes from, isn't it more likely to be something like
```python
if atom_num >= charge_ceil:
    density_ceil = numpy.array([charge_ceil])
```
?

Reply:
Yes. Thanks for catching this!